### PR TITLE
Add assignment 1 - week 1

### PR DESCRIPTION
**Pull request by baonq-me**

### CODE REVIEW

File: src/week1/calculator.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package main
+
+func eval(a, b float64, op func(a, b float64) (r float64, err error)) (r float64, err error) {
```
Comment (xuanit) [inline src/week1/calculator.go]: we could define a type for this kind of function `func(a, b float64) (r float64, err error)`.

File: src/week1/operator_map.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package main
+
+var (
+	fpMap = initOperatorMap()
+)
+
+func initOperatorMap() map[string]func(a, b float64) (r float64, err error) {
+
+	fpMap := make(map[string]func(a, b float64) (r float64, err error))
+	fpMap["+"] = add
```
Comment (xuanit) [inline src/week1/operator_map.go]: fyi: we could use map literal here. for example,
```
commits := map[string]int{
    "rsc": 3711,
    "r":   2138,
    "gri": 1908,
    "adg": 912,
}
```

File: src/week1/calculator/main.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+	scanner := bufio.NewScanner(os.Stdin)
+
+	fmt.Print("> ")
+
+	for scanner.Scan() {
+
+		text := scanner.Text()
+
+		if verify(text) {
```
Comment (hieuvo) [inline src/week1/calculator/main.go:21]: please apply `early return` or `early escape`

File: src/week1/calculator/verify.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package calculator
+
+import "regexp"
+
+var (
+	regex, _ = regexp.Compile(`^\-?\d+.?\d*\ +(\+|\-|\*|\/){1}\ \-?\d+.?\d*$`)
```
Comment (hieuvo) [inline src/week1/calculator/verify.go:6]: - `{1}` seems unnecessary
- `1. + 12.` will be passed by this regex
- multiple spaces support would be great

File: .idea/dictionaries/quocbao.xml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<component name="ProjectDictionaryState">
```
Comment (hieuvo) [inline .idea/dictionaries/quocbao.xml:1]: please use `.gitignore` to exclude unnecessary files

**Review by hieuvo — COMMENTED** [review]
btw, this submission has very high quality, you did a good job!

File: src/week1/calculator/verify.go (hunk 4 above)
Comment (baonq-me) [reply src/week1/calculator/verify.go:6]: - I think `1. + 12. ` is okay :D
- `\ +` support multiple spaces

File: .idea/dictionaries/quocbao.xml (hunk 5 above)
Comment (baonq-me) [reply .idea/dictionaries/quocbao.xml:1]: This is my intention because I use two laptops

**Pull request merged**